cluster/v1alpha1/types_addonplacementscore.go

```diff
@@ -0,0 +1,69 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:resource:scope="Namespaced"
+// +kubebuilder:subresource:status
+
+// AddOnPlacementScore represents a bundle of scores of one managed cluster, which could be used by placement.
+// AddOnPlacementScore is a namespace scoped resource. The namespace of the resource is the cluster namespace.
+type AddOnPlacementScore struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Status represents the status of the AddOnPlacementScore.
+	// +optional
+	Status AddOnPlacementScoreStatus `json:"status,omitempty"`
+}
+
+//AddOnPlacementScoreStatus represents the current status of AddOnPlacementScore.
+type AddOnPlacementScoreStatus struct {
+	// Conditions contain the different condition statuses for this AddOnPlacementScore.
+	// +optional
```

Comment:
It will be better if we add listmapkey and listmap marker here.

Reply:
Add listType and listMapKey.